erts/preloaded/src/erlang.erl

```diff
@@ -821,6 +833,16 @@ error(_Reason) ->
 error(_Reason, _Args) ->
     erlang:nif_error(undefined).
 
+%% error/3
+%% Shadowed by erl_bif_types: erlang:error/3
+-spec error(Reason, Args, Options) -> no_return() when
+      Reason :: term(),
+      Args :: [term()],
+      Options :: [Option],
+      Option :: {'error_info', #{'reason' := term()}}.
```

Comment:
My Dialyzer is not great but should this be:

```suggestion
      Option :: {'error_info', #{'reason' := term(), any() := any()}}.
```

?

Also, is there a reason (ha) to require a `reason` key? The `format_error/4` written so far all do `maps:get(reason, Map, none)`, which implies reason is not being strictly required.

Reply:
> Also, is there a reason (ha) to require a `reason` key?

Not really. I will talk with the team and perhaps change it.

Changing the spec in the way you suggested would allow a library to pass arbitrary keys and values in the map that its own `format_error/4` function would understand. That could be nice, but I am not sure that is a feature that is needed. 

Our reason for using a map was that we could add additional keys to the map that `erl_error` would know about (updating the spec at the same time).

Reply:
I see: it only allows the `reason` and nothing more. It makes sense to me now, thanks.